src/Utils/ProductGalleryUtils.php

```diff
@@ -6,6 +6,8 @@
  * {@internal This class and its methods are not intended for public use.}
  */
 class ProductGalleryUtils {
+	const CROP_IMAGE_SIZE = 500;
```

Comment:
Yes I thought about doing this but quickly dismissed it only because there is only so much space when displaying the main image and if we cropped it to a bigger resolution, then performance wise it is a hit because you would then be loading a much larger image into a smaller container which is not needed. But I am not oppose to doing it if you think that is better.

Reply:
> Yes I thought about doing this but quickly dismissed it only because there is only so much space when displaying the main image and if we cropped it to a bigger resolution, then performance wise it is a hit because you would then be loading a much larger image into a smaller container which is not needed. But I am not oppose to doing it if you think that is better.

I believe that WP generates multiple sizes for the same image and it will wisely choose the correct one based on the screen size, like we have in the `srcset` attribute: 
![CleanShot 2023-10-27 at 17 40 50@2x](https://github.com/woocommerce/woocommerce-blocks/assets/20469356/71abcbb0-bb9e-4d5a-b3b3-13a903b2afe7)

> Correct, I didn't implement it for the dialog because I thought it is supposed to work this way so shoppers can see the entire image when explicitly clicked to see a larger size. But happy to make the change if you think we should.

The dialog contains an independent instance of the Product Gallery block, which means that users can customize how they want the block to be displayed inside the dialog. For example, users can set the "crop image" option to 'true' in the Single Product template but have it set to 'false' in the Product Gallery template part (the dialog). So, in this case, we can default it to false in the dialog; however, the functionality for cropping images should be available in both places.

Reply:
Ok so then in this case, it should work as expected since the dialog itself has it's own set of options. Another words if you go to Patterns->Template Parts->Product Gallery you can turn on crop image setting and it would essentially enable it in the dialog.